tests/test_concurrent_simulation.py

```diff
@@ -58,47 +58,50 @@ def sim2():
     sim.addMolecules(spGreen, 20, highpos=[10, 50, 50])
 
     # Add Surfaces
-    r1 = smoldyn.Rectangle(corner=[100, 0, 0], dimensions=[100, 100], axis="-x")
-    r2 = smoldyn.Rectangle(corner=[0, 0, 0], dimensions=[100, 100], axis="+y")
-    r3 = smoldyn.Rectangle(corner=[0, 100, 0], dimensions=[100, 100], axis="-y")
-    r4 = smoldyn.Rectangle(corner=[0, 0, 0], dimensions=[100, 100], axis="+z")
-    r5 = smoldyn.Rectangle(corner=[0, 0, 100], dimensions=[100, 100], axis="-z")
+    r1 = sim.addRectangle(corner=[100, 0, 0], dimensions=[100, 100], axis="-x")
+    r2 = sim.addRectangle(corner=[0, 0, 0], dimensions=[100, 100], axis="+y")
+    r3 = sim.addRectangle(corner=[0, 100, 0], dimensions=[100, 100], axis="-y")
+    r4 = sim.addRectangle(corner=[0, 0, 0], dimensions=[100, 100], axis="+z")
+    r5 = sim.addRectangle(corner=[0, 0, 100], dimensions=[100, 100], axis="-z")
 
     s1 = sim.addSurface("walls", panels=[r1, r2, r3, r4, r5])
     # s1.both.setStyle(drawmode='edge')
     s1.setStyle("both", drawmode="edge")
     s1.setAction("both", [spRed, spGreen], "reflect")
 
     # portsurf
-    rr = smoldyn.Rectangle(corner=[0, 0, 0], dimensions=[100, 100], axis="+x")
+    rr = sim.addRectangle(corner=[0, 0, 0], dimensions=[100, 100], axis="+x")
     portSurf = sim.addSurface("portsurf", panels=[rr])
     portSurf.setStyle("front", drawmode="face", color="gray")
     portSurf.setStyle("back", drawmode="face", color=[0.2, 0, 0, 1])
     portSurf.setAction("front", [spRed, spGreen], "port")
     portSurf.setAction("back", [spRed, spGreen], "reflect")
 
     # Ports
-    smoldyn.Port("testport", surface=portSurf, panel="front")
-    # FIXME:
-    # sim.setGraphics("opengl", 20)
-    sim.addOutputData("data")
-    sim.addCommand(cmd="molcount data", cmd_type='E')
-    return sim
+    sim.addPort(name="testport", surface=portSurf, panel="front")
 
+    #  sim.setGraphics("opengl", 20)
+    sim.addOutputData("data2")
+    sim.addCommand(cmd="molcount data2", cmd_type='E')
+    return sim
 
 def main():
-    s1 = sim1()
-    s2 = sim2()
-    assert s1.simptr != s2.simptr
-    assert s1 != s2
-    t = time.time()
-    s1.run(100, dt=0.01, overwrite=True)
-    print('[INFO] First simulation is over.')
-    print(s1.getOutputData('data'))
-    s2.run(100, dt=0.01, overwrite=True)
-    print('[INFO] Second simulation is over.')
-    print(f"[INFO] Took time {time.time()-t} sec")
-    print(s2.getOutputData('data'))
+    s1, s2 = sim1(), sim2()
```

Comment:
This is the major accomplishment in this PR.

Two simulations can be run in two different threads.